sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -950,9 +950,15 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
       case f: Filter =>
         val (correlated, _) = splitConjunctivePredicates(f.condition).partition(containsOuter)
 
-        // Find any non-equality correlated predicates
+        // Find any non-equality correlated predicates and equality predicates that do not
+        // guarantee one-on-one mapping between inner and outer attributes. E.G:
+        // a = outer(c) -> true
+        // a > outer(c) -> false
+        // a + b = outer(c) -> false (because there can be multiple combinations of a, b that
+        // satisfy the condition)
         foundNonEqualCorrelatedPred = foundNonEqualCorrelatedPred || correlated.exists {
-          case _: EqualTo | _: EqualNullSafe => false
+          case Equality(_: Attribute, b) => b.find(_.isInstanceOf[Attribute]).isDefined
```

Comment:
Doesn't this also match `a = outer(b)`?

Reply:
`OuterReference` is a leaf node, so the code here means `b` can only contain outer references and literals.

@allisonwang-db can you add some comments to explain it?

Reply:
Oh, I missed `OuterReference` is a leaf node. Yea, it is nicer to add comment here.

Reply:
Will do!